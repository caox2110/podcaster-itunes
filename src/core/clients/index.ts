export { default as HttpClientFactory } from './HttpClient.factory';
export * from './enums';
export * from './types';
