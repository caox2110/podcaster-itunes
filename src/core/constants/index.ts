import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

/**
 * Constants for the app
 *
 * @type {{ app: any; theme: any; url: any; }}
 */
const constants = {
  app: appConstant,
  theme: themeConstant,
  url: urlConstant,
  queryClient: queryClientConstant,
};

if (appConstant.IS_DEV) {
  console.warn('Constants are', constants);
  console.warn(`ENV is ${appConstant.MODE}`);
}

export { constants };
