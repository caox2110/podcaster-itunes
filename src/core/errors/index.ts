export { default as ExceptionHandler } from './Exceptions.handler';
