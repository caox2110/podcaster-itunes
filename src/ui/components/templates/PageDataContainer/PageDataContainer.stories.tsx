import type { ComponentStory, ComponentMeta } from '@storybook/react';
import { withPerformance } from 'storybook-addon-performance';

import PageDataContainer from '.';

const Template: ComponentStory<typeof PageDataContainer> = (properties) => (
  <PageDataContainer {...properties} />
);

const PageDataContainerLoading = Template.bind({});
PageDataContainerLoading.args = {
  children: <h1>Testing Text</h1>,
  isLoading: true,
  isError: false,
  error: 'Error fetching podcast-rss. Message: Internal Server Error.',
};

const PageDataContainerError = Template.bind({});
PageDataContainerError.args = {
  children: <h1>Testing Text</h1>,
  isError: true,
  isLoading: false,
  error: 'Error fetching podcast-rss. Message: Internal Server Error.',
};

const PageDataContainerNormal = Template.bind({});
PageDataContainerNormal.args = {
  children: <h1>Testing Text</h1>,
  isError: false,
  isLoading: false,
  error: 'Error fetching podcast-rss. Message: Internal Server Error.',
};

export { PageDataContainerLoading, PageDataContainerError, PageDataContainerNormal };
export default {
  title: 'UI/Templates/PageDataContainer',
  component: PageDataContainer,
  parameters: {
    layout: 'fullscreen',
  },
  decorators: [withPerformance],
} as ComponentMeta<typeof PageDataContainer>;
