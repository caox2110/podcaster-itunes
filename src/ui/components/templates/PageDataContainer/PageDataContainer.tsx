import { ReactNode } from 'react';

import { ErrorContainer, Loading, SuspenseCustom } from '../..';

type PageDataContainerType = {
  isLoading: boolean;
  isError: boolean;
  error: string | undefined;
  children?: ReactNode;
};

const defaultProperties = { children: Element };

function PageDataContainer({ isLoading, isError, error, children }: PageDataContainerType) {
  if (isError) return <ErrorContainer error={error || ''} />;

  return <SuspenseCustom>{isLoading ? <Loading /> : children}</SuspenseCustom>;
}

PageDataContainer.defaultProps = defaultProperties;

export type { PageDataContainerType };
export default PageDataContainer;
