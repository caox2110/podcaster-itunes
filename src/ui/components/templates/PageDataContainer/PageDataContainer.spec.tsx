import { render } from '@/core/test';
import type { RenderResult } from '@/core/test';

import PageDataContainer, { PageDataContainerType } from './PageDataContainer';

const properties = {
  isLoading: false,
  isError: false,
  error: undefined,
};

function renderApp({ ...rest }: PageDataContainerType = properties): RenderResult {
  const utils = render(
    <PageDataContainer {...rest}>
      <h1>Testing text</h1>
    </PageDataContainer>,
  );
  return {
    ...utils,
  };
}

describe('PageDataContainer', () => {
  describe('Rendering', () => {
    it('Should render without throwing any errors', () => {
      const { baseElement, container } = renderApp();

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
