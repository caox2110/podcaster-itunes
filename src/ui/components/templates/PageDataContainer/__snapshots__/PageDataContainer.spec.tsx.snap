// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageDataContainer Rendering Should render without throwing any errors 1`] = `
<div>
  <h1>
    Testing text
  </h1>
</div>
`;
